editor/filesystem_dock.cpp

```diff
@@ -1172,6 +1172,20 @@ void FileSystemDock::_update_project_settings_after_move(const Map<String, Strin
 			}
 		};
 	}
+
+	// Also search for the file in autoload, as they are stored differently from normal files.
+	List<PropertyInfo> property_list;
+	ProjectSettings::get_singleton()->get_property_list(&property_list);
+	for (const List<PropertyInfo>::Element *E = property_list.front(); E; E = E->next()) {
+		if (E->get().name.begins_with("autoload/")) {
+			String autoload = GLOBAL_GET(E->get().name);
+			// Autoload resource paths are stored with a leading "*".
```

Comment:
`*` indicates that the autoload is available as a Singleton. Autoloads which don't have the "Singleton" box checked in ProjectSettings don't have a `*` in their path.

Reply:
Indeed, so you should handle both cases.

Reply:
Thanks, done! That's an interesting mechanism... I wonder if it would be more flexible / intuitive to simply store the value separately in project.godot? E.g.

```
[autoload]
some_autoload="res://script/some_script.gd"
some_autoload/singleton=true

; Or

some_autoload={
"path": "res://scripts/some_script.gd",
"singleton": true
}
```

But that's a fix for another day, I suppose.